src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -1206,7 +1205,11 @@ impl Coordinator {
                 // Emit notices.
                 self.emit_optimizer_notices(session, &global_lir_plan.df_meta().optimizer_notices);
 
-                let df_desc = global_lir_plan.unapply().0;
+                // Timestamp selection
+                let since = self.least_valid_read(&global_lir_plan.id_bundle(cluster_id));
```

Comment:
It's a bit weird that we need to resolve the timestamp before calling `GlobalLirPlan::unapply()`.

Can you add a `TODO` to all `pub fn id_bundle` methods define this once on `DataflowDescription` and use delegate calls in the `optimize` methods? The main obstacle at the moment is that `CollectionIdBundle` is defined in `adapter` and we first have to move it to `mz_compute_types`.

Reply:
We don't need to resolve the timestamp before calling unapply, doing it the other way round would work just as well using `dataflow_imports_id_bundle`. I can change it if you prefer.

Reply:
> I can change it if you prefer.

Yes please and if I am not mistaken you can just simplify all `pub fn id_bundle()` definitions to implement themselves using a `dataflow_import_id_bundle(...)` call.